### ISSUE TITLE
Launching FSI results in an error

### ISSUE THREAD

**Issue by 7sharp9** [ionide/ionide-vscode-fsharp#269]
<img width="1103" alt="screen shot 2016-11-06 at 12 29 49" src="https://cloud.githubusercontent.com/assets/588746/20037680/c52f10ac-a41c-11e6-9c3d-20c6593f9cca.png">

Logs are as follows:
```
WARNING: Promise with no error callback:85
winjs.base.js:46Objecterror: Error: pr.catch is not a function
    at Object.onFail (/Users/7sharp9/.vscode/extensions/Ionide.Ionide-fsharp-2.8.10/fable_external/Helpers-729269674.js:73:24)
    at /Users/7sharp9/.vscode/extensions/Ionide.Ionide-fsharp-2.8.10/Components/Fsi.js:97:30
    at /Users/7sharp9/.vscode/extensions/Ionide.Ionide-fsharp-2.8.10/fable_external/Helpers-729269674.js:35:15
    at Object.create (/Users/7sharp9/.vscode/extensions/Ionide.Ionide-fsharp-2.8.10/fable_external/Helpers-729269674.js:32:16)
    at start (/Users/7sharp9/.vscode/extensions/Ionide.Ionide-fsharp-2.8.10/Components/Fsi.js:75:25)
    at /Users/7sharp9/.vscode/extensions/Ionide.Ionide-fsharp-2.8.10/Components/Fsi.js:255:16
    at t.$executeContributedCommand (/Users/7sharp9/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/node/extensionHostProcess.js:10:25694)
    at t.e.handle (/Users/7sharp9/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/node/extensionHostProcess.js:10:10602)
    at s (/Users/7sharp9/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/node/extensionHostProcess.js:7:21356)
    at p (/Users/7sharp9/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/node/extensionHostProcess.js:7:22033)exception: nullhandler: undefinedid: 85parent: undefinedpromise: n.Class.derive._oncancel__proto__: Object
messageService.ts:128pr.catch is not a functione.doShow @ messageService.ts:128
messageService.ts:128Failed to spawn FSI, please ensure it's in PATHe.doShow @ messageService.ts:128
```

OSX with mono 4.6.1

**Comment by 7sharp9** [comment]
It fails at this point:

<img width="1458" alt="screen shot 2016-11-06 at 13 17 08" src="https://cloud.githubusercontent.com/assets/588746/20038004/62316138-a423-11e6-9137-8dbb29af8579.png">

Looks like `this` is the promise and not `pr` hence the: Error: pr.catch is not a function

**Comment by alfonsogarciacaro** [comment]
Thanks a lot for the information @7sharp9. I'd have to debug this to see what's going on. Another solution could be to use the Promise helpers in [fable-powerpack](https://github.com/fable-compiler/fable-powerpack/blob/master/src/Promise.fs), which have unit tests and seem to be working fine. I can send a PR to replace them if @Krzysztof-Cieslak wants. fable-powerpack relies on latest fable-compiler alpha to be used as a lib, but we can also just put the code in ionide-helpers.

**Comment by Krzysztof-Cieslak** [comment]
Should be fixed in `2.9.0`
